applications/nrf_desktop/doc/dfu.rst

```diff
@@ -77,6 +77,24 @@ These values are fetched using the :ref:`devinfo <dfu_devinfo>` configuration ch
    By default, the reported Vendor ID, Product ID, and generation are aligned with the values defined globally for the nRF Desktop application.
    The default values of Kconfig options used by the DFU module are based on respectively :ref:`CONFIG_DESKTOP_DEVICE_VID <config_desktop_app_options>`, :ref:`CONFIG_DESKTOP_DEVICE_PID <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_DEVICE_GENERATION <config_desktop_app_options>`.
 
+Flash access synchronization with other DFU methods
+===================================================
+
+The DFU module leverages the :ref:`nrf_desktop_dfu_lock` to synchronize flash access with other DFU methods (for example, SMP DFU).
+Other DFU methods must also use the :ref:`nrf_desktop_dfu_lock` for an effective synchronization mechanism.
+If multiple DFU transports are enabled in your application configuration, make sure that the following conditions are met:
+
+* The :ref:`CONFIG_DESKTOP_DFU_LOCK <config_desktop_app_options>` option is enabled
+* All of the used DFU transports use the :ref:`nrf_desktop_dfu_lock` module.
+
+On each DFU attempt, the module attempts to claim ownership over the DFU flash using the DFU Lock API.
+It holds the DFU owner status until the DFU process is completed or timed out.
+The lock is also kept during the background erase operation as it changes the DFU flash memory content.
+
+If the module is not the current DFU owner, it desists from performing requested actions that either write to the DFU flash or reboot the device.
+
+The module also automatically starts the background erase operation of the DFU flash once it becomes the lock owner after another DFU method.
+
```

Comment:
We may also update the `nrf_desktop_ble_smp` references in documentation. Currently it would be better to use our application-specific module because it improves synchronization (may be done in a separate doc improvement PR).

Reply:
moved to follow-up